### PR TITLE
Feature/issue 514

### LINKED ISSUE #514: Add issue deadline to calendar
When the assigned user receives the notification about the deadline for an issue, please provide the link to add to the user calendar.

### Why
It would be amazing if we can add the deadline to the user calendar, so this user will be aware of the deadline expected to send a solution to that issue.

### Level
Beginner

### Current behavior
When an email is sent to the assigned user, the user receives the email with the deadline date.

### Expected behavior
When an email is sent to the assigned user, the user will have the link on the email content to add that deadline to the calendar

### How to reproduce the user scenario
 - Create a user
 - Import an issue
 - Set deadline
 - Create another user
 - Apply to solve that issue
 - Log with the first user
 - Assign that user
 - The email notification should have the link to add to the calendar

### What you will need to complete this issue
You will need the development keys for Sendgrid to receive an email, that will be provided when you're assigned.

### Tips
Check the variables that we send on emails, so we need to add a new one with the link passing the event date to the link that points to the calendar.

### PR DESCRIPTION
**Pull request by Faithfinder**
## Description

> Adds a link to add the deadline as a Google Calendar event to the "Assigned" date

## Issue

> Closes #514

## Impacted Area

> Mail

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

> * Create a user
> * Import an issue
> * Set deadline
> * Create another user
> * Apply to solve that issue
> * Log with the first user
> * Assign that user
> * The email notification should have the link to add to the calendar

image

image

### CODE REVIEW

**Comment by Faithfinder** [comment]
I've used google translate for Portuguese locale, so I don't know how correct that is

**Comment by alexanmtz** [comment]
Great @Faithfinder! Thanks for all the PR and the details provided. I just need now you solve the conflicts with the main master, there's a conflict and once we solve we can merge 👍

**Comment by Faithfinder** [comment]
Done :)

File: modules/mail/assign.js [hunk 1]
```diff
@@ -210,4 +213,8 @@ if (constants.canSendEmail) {
   }
 }
 
+function determineCalendarLink (task) {
```
Comment (alexanmtz) [inline modules/mail/assign.js:216]: @Faithfinder, why not a modern approach?

```suggestion
const determineCalendarLink = (task) => {
```

Comment (Faithfinder) [reply modules/mail/assign.js:216]: Both approaches are used around the codebase. I don't really have a preference, though I wouldn't say that function declarations aren't modern. There are hoisting differences, and using declaration style allows you to not think about whether the order of lines is acceptable in every particular case.

**Pull request merged**